### PR TITLE
Fix graying out of unused members in LSP

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://devdiv.visualstudio.com/DevDiv/_queries/edit/2086491

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
@sharwell i cna't seem to figure out how to update the tests to pass.   they seem to eb validating spans in strange ways.

**Review by dibarbet — APPROVED** [review]

File: src/Analyzers/CSharp/Tests/RemoveUnusedMembers/RemoveUnusedMembersTests.cs [hunk 1]
```diff
@@ -3176,8 +3176,8 @@ class C
                 private C(int i) { }
```
Comment (DoctorKrolic) [inline src/Analyzers/CSharp/Tests/RemoveUnusedMembers/RemoveUnusedMembersTests.cs:3176]:
```suggestion
                {|#0:private C(int i) { }|}
```

File: src/Analyzers/CSharp/Tests/RemoveUnusedMembers/RemoveUnusedMembersTests.cs [hunk 2]
```diff
@@ -3176,8 +3176,8 @@ class C
...
             }
             """,
-// /0/Test0.cs(3,13): info IDE0051: Private member 'C.C' is unused
-VerifyCS.Diagnostic("IDE0051").WithSpan(3, 13, 3, 14).WithArguments("C.C"));
+            // /0/Test0.cs(3,13): info IDE0051: Private member 'C.C' is unused
+            VerifyCS.Diagnostic("IDE0051").WithSpan(3, 5, 3, 25).WithArguments("C.C"));
```
Comment (DoctorKrolic) [inline src/Analyzers/CSharp/Tests/RemoveUnusedMembers/RemoveUnusedMembersTests.cs:3180]:
```suggestion
            VerifyCS.Diagnostic("IDE0051").WithLocation(0).WithArguments("C.C"));
```

Comment (CyrusNajmabadi) [reply src/Analyzers/CSharp/Tests/RemoveUnusedMembers/RemoveUnusedMembersTests.cs]: that doesn't seem to fix things.

File: src/Analyzers/CSharp/Tests/RemoveUnusedMembers/RemoveUnusedMembersTests.cs (hunk 1 above)
Comment (CyrusNajmabadi) [reply src/Analyzers/CSharp/Tests/RemoveUnusedMembers/RemoveUnusedMembersTests.cs]: that doesn't seem to help.

File: src/Analyzers/Core/Analyzers/RemoveUnusedMembers/AbstractRemoveUnusedMembersDiagnosticAnalyzer.cs [hunk 3]
```diff
@@ -45,7 +45,7 @@ internal abstract class AbstractRemoveUnusedMembersDiagnosticAnalyzer<
         EnforceOnBuildValues.RemoveUnusedMembers,
         new LocalizableResourceString(nameof(AnalyzersResources.Remove_unused_private_members), AnalyzersResources.ResourceManager, typeof(AnalyzersResources)),
         new LocalizableResourceString(nameof(AnalyzersResources.Private_member_0_is_unused), AnalyzersResources.ResourceManager, typeof(AnalyzersResources)),
-        hasAnyCodeStyleOption: false, isUnnecessary: true);
+        hasAnyCodeStyleOption: false, isUnnecessary: false);
```
Comment (CyrusNajmabadi) [inline src/Analyzers/Core/Analyzers/RemoveUnusedMembers/AbstractRemoveUnusedMembersDiagnosticAnalyzer.cs:48]:
```suggestion
        hasAnyCodeStyleOption: false, isUnnecessary: true);
```

File: src/Analyzers/Core/Analyzers/RemoveUnusedMembers/AbstractRemoveUnusedMembersDiagnosticAnalyzer.cs [hunk 4]
```diff
@@ -54,7 +54,7 @@ internal abstract class AbstractRemoveUnusedMembersDiagnosticAnalyzer<
         EnforceOnBuildValues.RemoveUnreadMembers,
         new LocalizableResourceString(nameof(AnalyzersResources.Remove_unread_private_members), AnalyzersResources.ResourceManager, typeof(AnalyzersResources)),
         new LocalizableResourceString(nameof(AnalyzersResources.Private_member_0_can_be_removed_as_the_value_assigned_to_it_is_never_read), AnalyzersResources.ResourceManager, typeof(AnalyzersResources)),
-        hasAnyCodeStyleOption: false, isUnnecessary: true);
+        hasAnyCodeStyleOption: false, isUnnecessary: false);
```
Comment (CyrusNajmabadi) [inline src/Analyzers/Core/Analyzers/RemoveUnusedMembers/AbstractRemoveUnusedMembersDiagnosticAnalyzer.cs:57]:
```suggestion
        hasAnyCodeStyleOption: false, isUnnecessary: true);
```

**Comment by michaelcsikos** [comment]
This needs an option so we can disable it.

**Pull request merged**